### ISSUE TITLE
[Bug] /stopcasting & queued abilities in a macro

### ISSUE THREAD

**Issue by fantom8** [Atlantiss/NetherwingBugtracker#8589]
**Description**:
It should be possible to "toggle" skills that "queue" for the next auto attack.

The macro is something like this:
/cast Heroic Strike
/stopcasting

On first press the skill should be queued and on second press it should be canceled. This is how it functions on blizzard servers.

Example:
https://www.reddit.com/r/classicwow/comments/dnc2rp/fury_warrior_question_macro_heroic_strike_start/f59fod8/

More proof can be found on the Classic Warrior discord but it's kind of a pain to dig through that. If you feel like this would be a substantial change that requires more proof then I can look for more. But in my opinion this is a pretty small change that is almost entirely quality of life. This behavior can currently be mimicked with 2 buttons and a /cleartarget macro, but implementing /stopcasting would be a lot nicer for any druids or warriors on the server and will make the macro function in a more blizzlike manner.

**Current behaviour**: /stopcasting in a macro does not cancel a queued /cast action such as Heroic Strike, Cleave, or Maul

**Expected behaviour**: /stopcasting in a macro should cancel a queued /cast action such as Heroic Strike, Cleave, or Maul
Like this:
/cast Heroic Strike
/stopcasting

**Server Revision**: 3841

**Comment by Wolffenstein** [comment]
Your macro did not include /startattack from that reddit comment. Obviously if you just do 
/cast Heroic Strike
/stopcasting
The server will process each command in succession and if your melee swing is ready it will cast Heroic Strike before processing the /stopcasting command. There is no bug here. If you actually try to use the macro while your melee swing is not ready it will properly remove Heroic Strike from the queue. Everything works correctly here

**Comment by fantom8** [comment]
I see what you're saying, but I'm still not getting the behavior from the macro that I expect. I think that without a tiny batching window that perhaps this functionality cannot be replicated on the Netherwing core.

I have retested without /startattack, although I don't believe that line in a macro should affect any /cast behavior included or not. My normal macros do include /startattack and my initial testing for this macro behavior was done with /startattack included.

So without that there are two options for ordering the relevant lines of the macro:
/cast Heroic Strike
/stopcasting

and
/stopcasting
/cast Heroic Strike

TL;DR of results is: it seems like the server is *always* processing both lines. The first macro above will cast and immediately cancel HS. It will never activate the ability even if you only press it once. The second macro will always cast HS, even if it appears to "toggle off."

I have 2 evidence screenshots for this now. The first was made by pressing exactly this macro two times:
#showtooltip
/stopcasting
/cast Heroic Strike

Results 1: https://i.imgur.com/vxievAw.png
In between attacks I pressed the macro twice, then printscreen. After the next swing I pressed printscreen again and documented both screenshots. No other buttons were pressed between the attacks visible in the combat log.

The second was made by this macro (simply commented out the /stopcasting):
#showtooltip
--/stopcasting
/cast Heroic Strike

Results 2: https://i.imgur.com/42RC5tj.png
This is a control test. I followed the same process and took screenshots at the same times as the previous result.

As you can see they both have the same result of a Heroic Strike attack happening despite the HS cast appearing to have been "canceled" on my action bar on the first screenshot.

I would appreciate it if the behavior of one of the macros with /stopcasting could be adjusted so it either properly only cancels the HS if it was already active before the button press (order 1) *or* actually cancels the HS when it deactivates the button on the action bar (order 2). That is how both orders work in Classic. If that's impossible with how the server is coded then I understand though.
